ts/components/messages/Inbox.tsx

```diff
@@ -0,0 +1,225 @@
+import { fromNullable, none } from "fp-ts/lib/Option";
+import * as pot from "italia-ts-commons/lib/pot";
+import { Button, Text, View } from "native-base";
+import React from "react";
+import { FlatList, ListRenderItem, StyleSheet } from "react-native";
+
+import I18n from "../../i18n";
+import { lexicallyOrderedMessagesStateInfoSelector } from "../../store/reducers/entities/messages";
+import { MessageState } from "../../store/reducers/entities/messages/messagesById";
+import { paymentsByRptIdSelector } from "../../store/reducers/entities/payments";
+import { servicesByIdSelector } from "../../store/reducers/entities/services/servicesById";
+import InboxItem from "./InboxItem";
+
+export const styles = StyleSheet.create({
+  listWrapper: {
+    flex: 1
+  },
+  buttonBar: {
+    position: "absolute",
+    left: 0,
+    right: 0,
+    bottom: 0,
+    flexDirection: "row",
+    zIndex: 1,
+    justifyContent: "space-around",
+    backgroundColor: "#ddd",
+    padding: 10,
+    opacity: 0.75
+  },
+
+  buttonBarButton: {
+    opacity: 1
+  }
+});
+
+type Props = {
+  messagesStateInfo: ReturnType<
+    typeof lexicallyOrderedMessagesStateInfoSelector
+  >;
+  servicesById: ReturnType<typeof servicesByIdSelector>;
+  paymentsByRptId: ReturnType<typeof paymentsByRptIdSelector>;
+  refreshMessages: () => void;
+  setMessagesArchivedState: (
+    ids: ReadonlyArray<string>,
+    archived: boolean
+  ) => void;
+  onPressItem: (id: string) => void;
+};
+
+type State = {
+  lastMessageStatesUpdate: number;
+  filteredMessageStates: ReturnType<typeof generateFilteredMessageStates>;
+  isSelectionModeEnabled: boolean;
+  selected: Map<string, true>;
```

Comment:
```suggestion
  selectedMessageIds: Map<string, true>;
```

you can make this an `Option<Map<string, true>>` and get rid of `isSelectionModeEnabled`